### PR TITLE
Update LLVM 18.1.1

### PR DESCRIPTION
**Pull request by PiCoPress**
Update LLVM v18.1.1, and perhaps fixed submodule (src/config).

### CODE REVIEW

**Comment by sbc100** [comment]
So here we are updating from 18.1.0rc to 18.1.1?   (Can you mention both these versions in the PR title).  

I'm not sure why ever pinned to an `rc` version but its good that we are now back to an official release.

**Comment by alexcrichton** [comment]
I don't believe an rc was used, it was a [mistake in the LLVM release process](https://github.com/llvm/llvm-project/issues/84168).

File: .gitmodules [hunk 1]
```diff
@@ -7,3 +7,5 @@
 [submodule "src/config"]
 	path = src/config
 	url = https://git.savannah.gnu.org/git/config.git
+[submodule "src/config/"]
```
Comment (PiCoPress) [inline .gitmodules:10]: I found dup module definition at .gitmodules, so it should be removed.
For me, previous src/config repository has an issue which git can't fetch from, so I altered its mirror what yields expected result.
```...url/cgit/... -> ...url/git/....```

**Comment by PiCoPress** [comment]
Is there a benefit to accept LLVM 18.1.2?

**Comment by alexcrichton** [comment]
As a heads up I've opened https://github.com/WebAssembly/wasi-sdk/pull/402 to update to 18.1.2 which additionally addresses changes necessary to keep things working with components.

**Pull request closed without merging**